### DISCUSSION TITLE
[UI] Hiding properties from specific shaders in the UI

### DISCUSSION THREAD

**Discussion by panthavma** [bnpr/Malt#324] [Q&A] [answered]
Hello !

Sometimes, the UI for a material can get pretty bloated when you have a lot of parameters, especially if set by code.

![image](https://user-images.githubusercontent.com/22663538/172895907-764ccf4e-861e-4e31-9d9e-bfd04ab51ccc.png)

I was wondering if there was an easy way to either:
- Hide specific properties of shaders (like hiding nbPoints and all the points array in my example)
- Hide all the properties of a shader at once, like the rolling menus elsewhere in Blender.

Thanks for the answer!

**Comment by pragma37** [comment, marked as answer]
All-caps uniforms and uniforms that start with an underscore are hidden in the UI:

```GLSL
uniform bool _is_hidden = true;
uniform bool IS_HIDDEN = true;
```

There's currently no way to collapse all properties.

**Reply by panthavma** [reply]
Cool ! Thanks !
